### PR TITLE
Add liveness and readiness probes

### PR DESCRIPTION
**Pull request by kyguy**
Rolls and updates CC pod on liveness and readiness probe changes. Binds CC server to pod hostname and localhost

### CODE REVIEW

File: docker-images/kafka/cruise-control-scripts/cruise_control_config_generator.sh [hunk 1]
```diff
@@ -36,5 +36,6 @@ broker.metric.sample.store.topic=__KafkaCruiseControlModelTrainingSamples
 capacity.config.file=$CC_CAPACITY_FILE
 cluster.configs.file=$CC_CLUSTER_CONFIG_FILE
 webserver.accesslog.path=/tmp/access.log
+webserver.http.address=$(hostname)
```
Comment (ppatierno) [inline docker-images/kafka/cruise-control-scripts/cruise_control_config_generator.sh]: this is the address on which CC is listening for incoming HTTP requests right? Maybe we want something like the Kafka Connect REST API which is `$(hostname -I)`. Anyway we are hitting the CC REST API through the related service, right?

Comment (kyguy) [reply docker-images/kafka/cruise-control-scripts/cruise_control_config_generator.sh]: Yes and yes, I will update

Comment (kyguy) [reply docker-images/kafka/cruise-control-scripts/cruise_control_config_generator.sh]: Actually, not for the `$(hostname -I)`, The `webserver.http.address` field is meant to only take one address [1]. `$(hostname -I)` has the possibility of listing more than one IP address as the `-I` argument causes the command to "enumerates all configured addresses on all  network  interfaces"

[1] https://github.com/linkedin/cruise-control/blob/master/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlMain.java#L66

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java [hunk 2]
```diff
@@ -186,6 +211,35 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
...
+     * Estimates broker disk capacity from supplied storage configuration
+     *
+     * @param storage  Storage configuration for Kafka cluster
+     * @return  Broker disk capacity in MB as a String
+     */
+    private static String estimateDiskCapacityPerBrokerInMb(Storage storage) {
+        long capacity = 0;
+        switch (storage.getType()) {
+            case Storage.TYPE_JBOD:
+                capacity = parseMemory(((PersistentClaimStorage) storage).getSize()) / 1000000;
```
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java]: So what units is this capacity in? Are these in bytes (base 2) or iso (base 10)?

**Review by ppatierno — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java [hunk 3]
```diff
@@ -155,10 +177,13 @@ public static CruiseControl fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup ver
...
 
             cruiseControl = updateConfiguration(spec, cruiseControl);
-            KafkaConfiguration configuration = new KafkaConfiguration(kafkaSpec.getKafka().getConfig().entrySet());
+
+            KafkaClusterSpec kafkaClusterSpec = kafkaSpec.getKafka();
+            KafkaConfiguration configuration = new KafkaConfiguration(kafkaClusterSpec.getConfig().entrySet());
             if (configuration.getConfigOption(MIN_INSYNC_REPLICAS) != null) {
                 cruiseControl.minInsyncReplicas = configuration.getConfigOption(MIN_INSYNC_REPLICAS);
             }
+            cruiseControl.diskCapacityPerBroker = estimateDiskCapacityPerBrokerInMb(kafkaClusterSpec.getStorage());
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java]: is this related to liveness and readiness probes? If not, why it's in this PR? I would open a different one.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java (hunk 2 above)
Comment (kyguy) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java]: Base 10, but I am going to separate this commit into a different PR!

**Review by kyguy — COMMENTED** [review]
Stripped out the Broker capacity generation for a separate PR. Hope to piggyback some minor PR fixes with this PR though (But I can separate out as well if desired)l!

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java (hunk 2 above)
Comment (tomncooper) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java]: Ok but is that the same base CC uses?

Comment (kyguy) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java]: Yes base 10 for CC as well. Note that unlike the DISK capacity which is given in MB, the NW_IN and NW_OUT values are given in KB, and the CPU as a percentage

**Review by tomncooper — APPROVED** [review]
